leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/IntegrationTestHelper.java

```diff
@@ -246,6 +248,12 @@ public void deregisterClient() {
 
     public void dispose() {
         deregisterClient();
+        if (null != client) {
+            client.destroy(false);
+        }
+        if (null != server) {
+            server.destroy();
+        }
```

Comment:
It's not forbidden to call stop and additionally destroy.
But I will go to change the PR.